terraform/modules/common/instance/main.tf

```diff
@@ -0,0 +1,63 @@
+## Variables
+
+variable "stack_name" {
+  description = "The name of the stack within which the resource resides"
+}
+
+variable "ami_id" {
+  description = "The ami which will be used to build the instance"
+}
+
+variable "subnet_id" {
+  description = "The subnet which the instance will be added to"
+}
+
+variable "ssh_key_name" {
+  description = "The name of the ssh key which will assigned to the instance"
+}
+
+variable "security_groups" {
+  description = "A list of security groups which will be added to the instance. If the list is empty the default securit group will be used"
+  type        = "list"
+}
+
+variable "public_ip" {
+  description = "When this is true a public ip is added to the instance"
+  default     = false
+}
+
+## Resources
+
+resource "aws_instance" "instance" {
+  ami           = "${var.ami_id}"
+  subnet_id     = "${var.subnet_id}"
+  instance_type = "t2.small"
+
+  lifecycle {
+    ignore_changes = ["ami"]
+  }
+
+  root_block_device {
+    volume_size           = "10"
+    delete_on_termination = "true"
+  }
+
+  key_name = "${var.ssh_key_name}"
+
+  associate_public_ip_address = "${var.public_ip}"
+
+  security_groups             = ["${var.security_groups}"]
+  associate_public_ip_address = "${var.public_ip}"
+
+  tags {
+    Name      = "jumpbox.${var.stack_name}.dmz"
```

Comment:
Should we have something jumpbox specific in the general instance module?

Reply:
I think it's only used by the jumpbox project, so maybe could move the code into jumpbox?